### PR TITLE
Enable all default golingci-lint linters

### PR DESCRIPTION
**Pull request by cprivitere**
Update config to remove linters config section. This will enable all the default linters.
Fix linting problems found by all linters.

Could use some close scrutiny on my error handling fixes and my test fail log calls. 

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

### CODE REVIEW

File: metal/eip_controlplane_reconciliation.go [hunk 1]
```diff
@@ -148,8 +148,11 @@ func newControlPlaneEndpointManager(k8sclient kubernetes.Interface, stop <-chan
 			},
 		},
 	)
+	if err != nil {
+		return m, err
```
Comment (displague) [inline metal/eip_controlplane_reconciliation.go:152]: I'd argue we could return `nil, err`, but `m, err` is closer to the previous behavior. `m` will presumably be `nil` too.

File: metal/loadbalancers/metallb/configmap_test.go [hunk 2]
```diff
@@ -64,7 +64,9 @@ func TestConfigFileAddPeerByService(t *testing.T) {
 
 		cfg.Peers = peers[:]
 		peers := []Peer{tt.peer}
-		m.UpdatePeersByService(context.Background(), &peers, tt.addServiceNamespace, tt.addServicename)
+		if _, err := m.UpdatePeersByService(context.Background(), &peers, tt.addServiceNamespace, tt.addServicename); err != nil {
+			t.Fatalf("error updating peers by service")
```
Comment (displague) [inline metal/loadbalancers/metallb/configmap_test.go:68]:
```suggestion
			t.Fatalf("error updating peers by service: %s", err)
```

File: metal/loadbalancers/metallb/configmap_test.go [hunk 3]
```diff
@@ -152,7 +154,9 @@ func TestConfigFileRemovePeersByService(t *testing.T) {
 		}
 		m := &CMConfigurer{config: &cfg}
 
-		m.RemovePeersByService(context.Background(), tt.svcNamespace, tt.svcName)
+		if _, err := m.RemovePeersByService(context.Background(), tt.svcNamespace, tt.svcName); err != nil {
+			t.Fatalf("erorr removing peers by service")
```
Comment (displague) [inline metal/loadbalancers/metallb/configmap_test.go:158]:
```suggestion
			t.Fatalf("error removing peers by service")
```

File: metal/loadbalancers/metallb/configmap_test.go [hunk 4]
```diff
@@ -262,7 +266,9 @@ func TestConfigFileRemoveAddressPoolByAddress(t *testing.T) {
 		cfg.Pools = pools[:]
 		m := &CMConfigurer{config: &cfg}
 
-		m.RemoveAddressPoolByAddress(context.Background(), tt.addr)
+		if err := m.RemoveAddressPoolByAddress(context.Background(), tt.addr); err != nil {
+			t.Fatalf("error removing address pool")
```
Comment (displague) [inline metal/loadbalancers/metallb/configmap_test.go:270]:
```suggestion
			t.Fatalf("error removing address pool: %s", err)
```

File: metal/eip_controlplane_reconciliation.go (hunk 1 above)
Comment (cprivitere) [reply metal/eip_controlplane_reconciliation.go:152]: So...we're agreeing to leave it as m, err?

File: metal/loadbalancers/metallb/configmap_test.go [hunk 5]
```diff
@@ -64,7 +64,9 @@ func TestConfigFileAddPeerByService(t *testing.T) {
 
 		cfg.Peers = peers[:]
 		peers := []Peer{tt.peer}
-		m.UpdatePeersByService(context.Background(), &peers, tt.addServiceNamespace, tt.addServicename)
+		if _, err := m.UpdatePeersByService(context.Background(), &peers, tt.addServiceNamespace, tt.addServicename); err != nil {
```
Comment (ctreatma) [inline metal/loadbalancers/metallb/configmap_test.go:67]: Here we're making the method call _and_ checking for an error in the `if` condition, but in other places in this PR (e.g. [`metal/eip_controlplane_reconciliation.go`, line 155](https://github.com/equinix/cloud-provider-equinix-metal/pull/371/files#diff-e21bc796cf3fcaa8de618836c7670a9fe7bb512313c6328f63aa66f328e9d462R155) we're making the method call outside of the `if` condition.  Is this a style preference?  Or is it prescribed by the linter?

**Review by ctreatma — COMMENTED** [review]
I have one style question (not even a nit) and the DCO check is failing.  If we don't mind the DCO failure I'm good to approve this.

File: metal/loadbalancers/metallb/configmap_test.go (hunk 5 above)
Comment (cprivitere) [reply metal/loadbalancers/metallb/configmap_test.go:67]: Ok, I've swapped it over to the if _, err, style.

**Review by displague — APPROVED** [review]

**Pull request merged**